File: 45.jump-game-ii/45.py
class Solution:
    def jump(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if(len(nums) == 1):
            return 0
        index = 0
        jumpNumber = 1
        jumpFarthest = index + nums[index]
        newjumpFarthest = 0
        for index in range(1,len(nums)):
            if(jumpFarthest >= len(nums) - 1):
                return jumpNumber
            if(index <= jumpFarthest):
                tmp = index + nums[index]
                if(newjumpFarthest < tmp):
                    newjumpFarthest = tmp
            if(index == jumpFarthest):
                jumpFarthest = newjumpFarthest
                newjumpFarthest = 0
                jumpNumber  = jumpNumber + 1
        return jumpNumber

File: 45.jump-game-ii/45.js
/*
 * @lc app=leetcode id=45 lang=javascript
 *
 * [45] Jump Game II
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var jump = function(nums) {
    var jumpNumber = 0;
    var nextRange = 0 + nums[0];
    var nextnextRange = 0 + nums[0];
    for(var i = 1; i < nums.length; i++){
        if(nextRange >= nums.length - 1){
            return jumpNumber + 1;
        }      
        if(nextnextRange < i + nums[i]){
            nextnextRange = i + nums[i];
        }
        if(i == nextRange){
            nextRange = nextnextRange;
            jumpNumber ++;
        }
    }
    return jumpNumber
};
// @lc code=end

